modules/terraform/var-file_test.go

```diff
@@ -0,0 +1,250 @@
+		tags = {
+			foo = "bar"
+		}
+		list = ["item1"]`)
+
+	WriteFile(t, randomFileName, testHcl)
+	defer os.Remove(randomFileName)
+
+	val := GetVariableAsListFromVarFile(t, randomFileName, "list")
+
+	require.Equal(t, expected, val)
+}
+
+func TestGetVariableAsListNotList(t *testing.T) {
+	randomFileName := fmt.Sprintf("./%s.tfvars", random.UniqueId())
+
+	testHcl := []byte(`
+		aws_region     = "us-east-2"
+		aws_account_id = "111111111111"
+		tags = {
+			foo = "bar"
+		}
+		list = ["item1"]`)
+
+	WriteFile(t, randomFileName, testHcl)
+	defer os.Remove(randomFileName)
+
+	_, err := GetVariableAsListFromVarFileE(t, randomFileName, "tags")
+
+	require.Error(t, err)
+}
+
+func TestGetVariableAsListKeyDoesNotExist(t *testing.T) {
+	randomFileName := fmt.Sprintf("./%s.tfvars", random.UniqueId())
+
+	testHcl := []byte(`
+		aws_region     = "us-east-2"
+		aws_account_id = "111111111111"
+		tags = {
+			foo = "bar"
+		}
+		list = ["item1"]`)
+
+	WriteFile(t, randomFileName, testHcl)
+	defer os.Remove(randomFileName)
+
+	_, err := GetVariableAsListFromVarFileE(t, randomFileName, "badkey")
+
+	require.Error(t, err)
+}
+func TestGetAllVariablesFromVarFileEFileDoesNotExist(t *testing.T) {
+	var variables map[string]interface{}
+
+	err := GetAllVariablesFromVarFileE(t, "filea", variables)
+
+	require.Equal(t, "open filea: no such file or directory", err.Error())
+}
+
+func TestGetAllVariablesFromVarFileBadFile(t *testing.T) {
+	var variables map[string]interface{}
+
+	randomFileName := fmt.Sprintf("./%s.tfvars", random.UniqueId())
+	testHcl := []byte(`
+		thiswillnotwork`)
+
+	err := ioutil.WriteFile(randomFileName, testHcl, 0644)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		t.FailNow()
+	}
+
+	defer os.Remove(randomFileName)
+
+	err = GetAllVariablesFromVarFileE(t, randomFileName, variables)
+
+	if err == nil {
+		t.FailNow()
+	}
+
+	// HCL library could change their error string, so we are only testing the error string contains what we add to it
+	require.Regexp(t, fmt.Sprintf("^%s - ", randomFileName), err.Error())
+
+}
+
+func TestGetAllVariablesFromVarFile(t *testing.T) {
+	var variables map[string]interface{}
+
+	randomFileName := fmt.Sprintf("./%s.tfvars", random.UniqueId())
+	testHcl := []byte(`
+	aws_region     = "us-east-2"
+	`)
+
+	err := ioutil.WriteFile(randomFileName, testHcl, 0644)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		t.FailNow()
+	}
+
+	defer os.Remove(randomFileName)
+
+	err = GetAllVariablesFromVarFileE(t, randomFileName, &variables)
+
+	if err != nil {
+		t.FailNow()
+	}
+
+	expected := make(map[string]interface{})
+	expected["aws_region"] = "us-east-2"
+
+	require.Equal(t, expected, variables)
+
+}
+
+// Helper function to write a file to the filesystem
+// Will immediately fail the test if it could not write the file
+func WriteFile(t *testing.T, fileName string, bytes []byte) {
+	err := ioutil.WriteFile(fileName, bytes, 0644)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		t.FailNow()
+	}
```

Comment:
This will produce a more readable error message, and is more concise:

```suggestion
	require.NoError(t, err)
```